### ISSUE TITLE
Request: Add int8 support to Unsorted_Segment_X ops

### ISSUE THREAD

**Issue by sicong-li-arm** [tensorflow/tensorflow#81348]
(I also raised this request in the new LiteRT repo:  duplicate of https://github.com/google-ai-edge/LiteRT/issues/190)

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 24.04.1, but targeting Android
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (or github SHA if from source): 2.18.0

**Provide the text output from tflite_convert**: Used a custom converter that performs post training full quantization using representative datasets.

```python
converter = tf.lite.TFLiteConverter.from_keras_model(base_model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.representative_dataset = create_representative_data(data_path)
```

Hi! As part of my investigation of deploying a graph neural network (GNN), built from the [TF-GNN ](https://github.com/tensorflow/gnn) library on mobile (Android), I found that the operator [Unsorted_Segment_Sum](https://www.tensorflow.org/mlir/tfl_ops#tflunsorted_segment_sum_tflunsortedsegmentsumop) doesn't support int8. This prevented us from taking advantage of a full quantization of the GNN which relies on the unsorted_segment_x operators for core message passing steps. This resulted in quantized models less performant than the non-quantized ones because of the extra dequantization-quantization layers.

I'd like to request the addition of this data type, since it is a very important operator used by the TF-GNN library itself, and we may see more demand for quantization of GNNs in the future.

Thanks in advance!

**Comment by gaikwadrahul8** [comment]
Hi, @sicong-li-arm 
I apologize for the delayed response, I see at the moment we do support `int32` and `float32` data types but not `int8` data type so this issue will be considered as feature request and will update you 

https://github.com/tensorflow/tensorflow/blob/38f4882ba0a24bcdc2f1ce52be62e1b19242dbee/tensorflow/lite/kernels/unsorted_segment.cc#L205

Thank you for your cooperation and patience.

**Comment by gaikwadrahul8** [comment]
Hi, @pkgoogle 
Please take a look into this issue. Thank you.

**Comment by pkgoogle** [comment]
Please follow progress in LiteRT, thanks.
